### ISSUE TITLE
Problem comsuming WAGMI in custom package

### ISSUE THREAD

**Issue by 0xKingKoala** [wevm/wagmi#4314]
### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Describe the bug

I want to move components to a UI library so they can be re-used in my other apps.

#### Expected behavior
Using a component from my packages should use the `<WagmiProvider/>` in the tree

#### Actual behavior 
Using a component from my packages does not detect the `<WagmiProvider/>`, and gives the following error.

```
WagmiProviderNotFoundError: `useConfig` must be used within `WagmiProvider`.

Docs: https://wagmi.sh/react/api/WagmiProvider.html
Version: wagmi@2.12.16
Source

packages/ui/src/user-info-package.tsx (4:29) @ UserInfoPackage

  2 |
  3 | export function UserInfoPackage() {
> 4 |    const account = useAccount()
    |                             ^
  5 |   const {connectors, connect, status, error} = useConnect()
  6 |   const {disconnect} = useDisconnect()
  7 |
  ```

### Link to Minimal Reproducible Example

https://github.com/0xKingKoala/issue

### Steps To Reproduce

```
git clone git@github.com:0xKingKoala/issue.git
cd issue
pnpm install
pnpm dev
```

Navigate to the webpage, and the issue should present itself.

### What Wagmi package(s) are you using?

wagmi

### Wagmi Package(s) Version(s)

2.12.16

### Viem Version

2.21.17

### TypeScript Version

^5

### Anything else?

Repo is created using `turbo-repo create`, and the app was copied and pasted from the `wagmi-create next-app`

**Comment by tmm** [comment]
Most likely an issue with your setup (usually it is when someone is using turborepo).

**Comment by 0xKingKoala** [comment]
Fixed - See this commit [20540d909590cbb4a1d55718581ba60d02f9b0fd](https://github.com/0xKingKoala/issue/commit/20540d909590cbb4a1d55718581ba60d02f9b0fd)

Someone said to 
- add the `.npmrc` lines
- pnpm prune
- lock versions, and use same in both `package.json`
- add peerDeps for the WAGMI stuff.
